tfx/examples/cifar10/distributed/README.md

```diff
@@ -0,0 +1,86 @@
+# CIFAR-10 Transfer Learning and MLKit integration Example with Distributed Training
+
+This example illustrates how to modify the base example for distributed training with a distributed node pool on Google Kubernetes Engine(GKE).
+
+## Instructions
+
+### Set Up a Node Pool
+The guide assumes that you have command line access to `kubectl`.
+If you do not have access to a GKE cluster, follow the quickstart guide [here](https://cloud.google.com/kubernetes-engine/docs/quickstart) to start one.
+
+Then, follow the instructions below for how to set up a node pool on GKE:
+https://cloud.google.com/kubernetes-engine/docs/how-to/node-pools
+
+If you wish to use GPUs in your training, you may follow this guide instead:
+https://cloud.google.com/kubernetes-engine/docs/how-to/gpus
+
+### Set Up a Custom Image for Training (Optional)
+The base cifar10 example relies on dependencies including TensorFlowJS that do not come with the default TFX installation. 
+If you do not wish to use a custom docker image, you can remove the TFLite model rewriting portion in `cifar10_utils_native_keras`.
+
+Start from a base tensorflow gpu image in Docker:
+
+```
+docker run -it tensorflow/tensorflow:latest-gpu
+```
+
+Then, follow the instructions in the base example to configure your training environment. Once you are done, commit the image with:
+
+```
+docker commit [CONTAINER_ID] [new_image_name]
+```
+
+where the container id can be looked up via `docker ps -a`. Use
+
+```
+docker push [new_image_name]
+```
+
+to upload your new image to your container registry.
+
+### Modify Pipeline and Util Files for Distributed Training
+
+First, modify the `_cifar10_root` in `cifar10_pipeline_native_keras` to point to a remote directory (For example, `gs://YOUR_GCS_PATH`),
+and upload the cifar10 base directory there when you are done modifying the files. This allows training worker pods to pull the utility files
+from your remote directory during training.
+
+Then, replace the Trainer component with the `kubernetes_trainer_executor.GenericExecutor` imported from `tfx.extensions.google_cloud_kubernetes.trainer`:
```

Comment:
Then, specify the custom Trainer executor  `kubernetes_trainer_executor.GenericExecutor` ....